### PR TITLE
boot-timestamps: Discard firmware init time when running in a VM

### PR DESCRIPTION
**Pull request by medhefgo**
Fixes: #22060

### CODE REVIEW

File: src/shared/boot-timestamps.c [hunk 1]
```diff
@@ -33,12 +39,13 @@ int boot_timestamps(const dual_timestamp *n, dual_timestamp *firmware, dual_time
          * the monotonic timestamps here as negative of the actual
          * value. */
 
-        firmware->monotonic = y;
-        loader->monotonic = y - x;
-
-        a = n->monotonic + firmware->monotonic;
-        firmware->realtime = n->realtime > a ? n->realtime - a : 0;
+        if (discard_firmware) {
```
Comment (bluca) [inline src/shared/boot-timestamps.c]: Did you mean `!discard_firmware`?

Comment (medhefgo) [reply src/shared/boot-timestamps.c]: You're right. The perils of compile time testing. :D

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
e683878c0f should fix the test failures.

**Comment by poettering** [comment]
I am not sure I follow? Isn't this just a bug in qemu? Why tape over this?

What am I missing?

**Pull request merged**